src/modules/ssh_docker/mappers/docker_container.ts

```diff
@@ -0,0 +1,134 @@
+import Modem from 'docker-modem';
+import Docker, { ContainerInfo, DockerOptions } from 'dockerode';
+import _ from 'lodash';
+import SSH2Promise from 'ssh2-promise';
+
+import { Context, Crud, MapperBase } from '../../interfaces';
+import { sshAccounts } from '../../ssh_accounts';
+import { SshCredentials } from '../../ssh_accounts/entity';
+import { DockerContainer } from '../entity';
+import { SshDocker } from '../index';
+
+export class DockerContainerMapper extends MapperBase<DockerContainer> {
+  module: SshDocker;
+  entity = DockerContainer;
+  equals = (a: DockerContainer, b: DockerContainer) => {
+    const res =
+      Object.is(a.name, b.name) &&
+      Object.is(a.command, b.command) &&
+      Object.is(a.created.getTime(), b.created.getTime()) &&
+      _.isEqual(a.ports, b.ports) &&
+      _.isEqual(a.labels, b.labels) &&
+      Object.is(a.state, b.state) &&
+      Object.is(a.status, b.status) &&
+      _.isEqual(a.hostConfig, b.hostConfig) &&
+      _.isEqual(a.networkSettings, b.networkSettings) &&
+      _.isEqual(a.mounts, b.mounts);
+    return res;
+  };
+
+  private containerMapper(serverName: string, container: ContainerInfo): DockerContainer {
+    const out = new DockerContainer();
+    out.serverName = serverName;
+    out.containerId = container.Id;
+    out.name = container.Names[0];
+    out.image = container.Image;
+    out.imageId = container.ImageID;
+    out.command = container.Command;
+    out.created = new Date(container.Created * 1000);
+    out.ports = container.Ports;
+    out.labels = container.Labels;
+    out.state = container.State;
+    out.status = container.Status;
+    out.hostConfig = container.HostConfig;
+    out.networkSettings = container.NetworkSettings;
+    out.mounts = container.Mounts;
+    return out;
+  }
+
+  private async getDockerForServer(ctx: Context, serverName: string): Promise<Docker | undefined> {
+    const sshClient: SSH2Promise = await ctx.getSshClient(serverName);
+    const sshConfig = sshClient.config[0];
+    const modem = new Modem({
+      protocol: 'ssh',
+      host: sshConfig.host,
+      port: sshConfig.port,
+      username: sshConfig.username,
+      sshOptions: {
+        privateKey: sshConfig.privateKey,
+        passphrase: sshConfig.passphrase,
+      },
+    });
+    const docker = new Docker({ modem } as DockerOptions);
+    try {
+      await docker.ping();
+    } catch (e) {
+      return undefined;
+    }
+    return docker;
+  }
+
+  cloud = new Crud({
+    create: async (es: DockerContainer[], ctx: Context) => {
+      // create docker containers through RPC
```

Comment:
You should be able to create docker containers through this flow, I think. An RPC shouldn't be required.

Reply:
Also, I don't see the RPC you're talking about in this PR?